cmake/ProjUtilities.cmake

```diff
@@ -76,27 +76,38 @@ function(configure_proj_pc)
   set(datadir "$\{datarootdir\}")
   set(PACKAGE "proj")
   set(VERSION ${PROJ_VERSION})
-  # Build list for Libs.private
-  set(EXTRA_LIBS
-    -lstdc++
-    -lsqlite3
-    ${CMAKE_THREAD_LIBS_INIT}
-  )
+  # Build strings of dependencies (Libs.private, Requires.private)
+  string(REPLACE ";" " " EXTRA_LIBS "${CMAKE_THREAD_LIBS_INIT}")
+  set(EXTRA_REQUIRES "")
+  macro(add_module_or_libs MODULE LIBS)
+    find_package(PkgConfig QUIET)
+    pkg_check_modules(PROJ_PC_${MODULE} ${MODULE})
+    if(PROJ_PC_${MODULE}_FOUND)
+      string(APPEND EXTRA_REQUIRES " ${MODULE}")
+    else()
+      string(APPEND EXTRA_LIBS " ${LIBS}")
+    endif()
+  endmacro()
+  add_module_or_libs("sqlite3" "-lsqlite3")
   if(TIFF_ENABLED)
-    list(APPEND EXTRA_LIBS -ltiff)
+    add_module_or_libs("libtiff-4" "-ltiff")
```

Comment:
Can we somehow obtain the '-4' suffix from the dependency instead of hardcoding it here?

Reply:
The information simply is not available at this point, but add_module_or_libs checks that the module exists. 
Alternatives:
- Use an input (cache) variable.
- Attempt several names until hitting an existing/matching name.
- Use pkg-config to detect tiff, and use just that module.